src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -471,14 +471,19 @@ export function findPreferredKernel(
                 }
 
                 // Find a kernel spec that matches the language in the notebook metadata.
-                if (score <= 0 && speclanguage === (nbMetadataLanguage || '')) {
+                if (score <= 0 && nbMetadataLanguage && speclanguage === (nbMetadataLanguage || '')) {
```

Comment:
This is incorrect, we need to ensure we have a language, else we match anything that doesnt' have a language or for those where it cannot be determined.